sycl/include/sycl/multi_ptr.hpp

```diff
@@ -177,8 +186,8 @@ class multi_ptr {
   multi_ptr(accessor<typename std::remove_const_t<RelayElementType>, Dimensions,
                      Mode, access::target::device, isPlaceholder, PropertyListT>
                 Accessor)
-      : multi_ptr(
-            detail::cast_AS<decorated_type *>(Accessor.get_pointer().get())) {}
+      : multi_ptr(detail::cast_AS<decorated_type *>(
+            Accessor.template get_multi_ptr<DecorateAddress>().get())) {}
```

Comment:
```suggestion
      : multi_ptr(Accessor.template get_multi_ptr<DecorateAddress>()) {}
```

that's likely to insert an unnecessary cast otherwise

Reply:
Thanks @Naghasan. I addressed un-necessary casting using `get_decorated`. Your suggestion failed to work for a number of combinations of `address_space` and `decorated`.